infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -0,0 +1,485 @@
+import org.corfudb.protocols.wireprotocol.StreamAddressRange;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuCompactorManagement.ActiveCPStreamMsg;
+import org.corfudb.runtime.CorfuCompactorManagement.CheckpointingStatus;
+import org.corfudb.runtime.CorfuCompactorManagement.CheckpointingStatus.StatusType;
+import org.corfudb.runtime.CorfuCompactorManagement.StringKey;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.TableName;
+import org.corfudb.runtime.DistributedCompactor;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.CorfuStoreEntry;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.AbortCause;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+import org.corfudb.runtime.object.transactions.TransactionalContext;
+import org.corfudb.runtime.proto.RpcCommon;
+import org.corfudb.runtime.proto.service.CorfuMessage;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.TableRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * This class does all services that the coordinator has to perform. The actions performed by the coordinator are -
+ * 1. Call trimLog() to trim everything till the token saved in the Checkpoint table
+ * 2. Set CompactionManager's status as STARTED, marking the start of a compaction cycle.
+ * 3. Validate liveness of tables in the ActiveCheckpoints table in order to detect slow or dead clients.
+ * 4. Set CompactoinManager's status as COMPLETED or FAILED based on the checkpointing status of all the tables. This
+ *    marks the end of the compactoin cycle.
+ */
+
+public class CompactorLeaderServices {
+    private Table<StringKey, CheckpointingStatus, Message> compactionManagerTable;
+    private Table<TableName, CheckpointingStatus, Message> checkpointingStatusTable;
+    private Table<TableName, ActiveCPStreamMsg, Message> activeCheckpointsTable;
+    private Table<StringKey, RpcCommon.TokenMsg, Message> checkpointTable;
+
+    private Map<TableName, Tuple<Tuple<Long, Long>, Long>> readCache = new HashMap<>();
+
+    private final CorfuRuntime corfuRuntime;
+    private final CorfuStore corfuStore;
+    private final String nodeEndpoint;
+
+    @Setter
+    private boolean isLeader;
+
+    @Setter
+    private long epoch;
+    private Logger syslog;
+
+    private static class LivenessValidatorHelper {
+        @Getter
+        @Setter
+        private static long prevIdleCount = -1;
```

Comment:
please use something more explicit. Also please try to avoid using setters as much as possible.
If you need some example what other approaches could be used here please let me know

Reply:
Removed setters.